src/internal.rs

```diff
@@ -459,11 +459,11 @@ pub struct Rav1dFrameContext_lf {
     pub cdef_buf_plane_sz: [c_int; 2], /* stride*sbh*4 */
     pub cdef_buf_sbh: c_int,
     pub lr_buf_plane_sz: [c_int; 2], /* (stride*sbh*4) << sb128 if n_tc > 1, else stride*4 */
-    pub re_sz: c_int,                /* h */
     pub lim_lut: Align16<Av1FilterLUT>,
     pub last_sharpness: c_int,
     pub lvl: [[[[u8; 2]; 8]; 4]; 8], /* [8 seg_id][4 dir][8 ref][2 is_gmv] */
-    pub tx_lpf_right_edge: [*mut u8; 2],
+    pub tx_lpf_right_edge_y: Vec<u8>, /* len = h */
+    pub tx_lpf_right_edge_uv: Vec<u8>, /* len = h */
```

Comment:
I'm not sure it helps so much to split them into separate `Vec`s.  It is a simple implementation, but I don't think keeping them contiguous, which clearly had some benefit, is too hard.  Can we do this?

```rust
impl Rav1dFrameContext_lf {
    pub fn tx_lpf_right_edge(&self) -> [&[u8]; 2] {
        let mid = self.tx_lpf_right_edge.len() / 2;
        let (y, uv) = self.tx_lpf_right_edge.split_at(mid);
        [y, uv]
    }

    pub fn tx_lpf_right_edge_mut(&mut self) -> [&mut [u8]; 2] {
        let mid = self.tx_lpf_right_edge.len() / 2;
        let (y, uv) = self.tx_lpf_right_edge.split_at_mut(mid);
        [y, uv]
    }
}
```

Then usages are the same as before (though we could use a tuple instead of a `[_; 2]`; either is fine).

Reply:
I gave it a shot and didn't run into any borrow issues, so seems like this works!